### ISSUE TITLE
Plotting attention map in a comprehensive way

### ISSUE THREAD

**Issue by zichunxx** [LukeDitria/pytorch_tutorials#2]
Hi! @LukeDitria 

Thanks for your great video and tutorials. 

I'm trying to understanding the method to plot the attention map with the below code

```py
with torch.no_grad():
    x = model.conv1(test_images[:8].to(device))
    _, att_map = model.use_attention(x)
    
# Index of the image you want to visualize
img_idx = 6

# Specify the dimensions for the attention map visualization
x_dim = 5
y_dim = 25

assert x_dim < test_images.shape[3], "x_dim must be less than " + str(test_images.shape[3] - 1)
assert y_dim < test_images.shape[2], "y_dim must be less than " + str(test_images.shape[2] - 1)

# Plot the image and its corresponding attention map
fig, axes = plt.subplots(1, 2, figsize=(6, 3))

# Plot the original image
img_out = test_images[img_idx]
img_out = (img_out - img_out.min())/(img_out.max() - img_out.min())
axes[0].imshow(img_out.permute(1, 2, 0).cpu().numpy())
axes[0].set_title("Original Image")
axes[0].axis('off')
axes[0].scatter(x_dim, y_dim, color='red', marker='x')

# Plot the attention map
axes[1].imshow(att_map[img_idx, x_dim * y_dim].reshape(32, 32).cpu().numpy(), cmap='viridis')
axes[1].set_title("Attention Map")
axes[1].axis('off')

plt.show()
```

, in which the `x_dim` and `y_dim` should be given to plot the specific row of the attention matrix. 

I wondered if plotting the whole attention map comprehensively with `torch.max`, ' torch.mean`, or any other metric is right.

Thanks in advance for checking this issue!

**Comment by LukeDitria** [comment]
Hi,
Yes for each pixel location you could take the expectation of the softmax score of every other location querying that location.

**Comment by zichunxx** [comment]
Thanks for your reply!

I'm not sure if I catch your point. This line 
```
axes[1].imshow(att_map[img_idx, x_dim * y_dim].reshape(32, 32).cpu().numpy(), cmap='viridis')
```
can be replaced by 
```
axes[1].imshow(att_map[img_idx].softmax(dim=-1).mean(dim=-1).reshape(32, 32).cpu().numpy(), cmap="viridis")
```
Is that right?

**Comment by LukeDitria** [comment]
Hi,
all you should have to do is this:

```
axes[1].imshow(att_map[img_idx, :, :].mean(0).reshape(32, 32).cpu().numpy(), cmap='viridis')
```

**Comment by zichunxx** [comment]
Thanks for your help!
